src/collector/custom_score_top_collector.rs

```diff
@@ -117,9 +117,9 @@ where
 
 impl<F, TScore> CustomSegmentScorer<TScore> for F
 where
-    F: 'static + Sync + Send + Fn(DocId) -> TScore,
+    F: 'static + Sync + Send + FnMut(DocId) -> TScore,
```

Comment:
Can we remove `Sync` here too?

Reply:
... and possibly Send.

Reply:
I don't really understand the ramifications of this but the tests pass.